src/main/java/frc/lib/swerve/MAXSwerve.java

```diff
@@ -0,0 +1,346 @@
+import edu.wpi.first.math.geometry.Twist2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveDriveOdometry;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.RunCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class MAXSwerve extends SubsystemBase {
+  public enum ModuleLocation {
+    frontLeft(0),
+    frontRight(1),
+    backLeft(2),
+    backRight(3);
+
+    public final int value;
+    private static final ModuleLocation[] m_mapping =
+        new ModuleLocation[] {frontLeft, frontRight, backLeft, backRight};
+
+    private ModuleLocation(int v) {
+      this.value = v;
+    }
+
+    public static ModuleLocation fromInt(int v) {
+      return m_mapping[v];
+    }
+  }
+  // Robot MAXswerve modules
+  private final MAXModule m_frontLeft;
+  private final MAXModule m_frontRight;
+  private final MAXModule m_backLeft;
+  private final MAXModule m_backRight;
+
+  // The gyro sensor
+  private final AHRS m_gyro;
+
+  // The module positions
+  private final SwerveModulePosition[] m_ModulePositions;
+
+  // Kinematics
+  private final SwerveDriveKinematics m_kinematics;
+
+  // Odometry
+  SwerveDriveOdometry m_odometry;
+  private final Field2d m_field = new Field2d();
+
+  private final double m_maxSpeed;
+
+  /**
+   * Create a Swerve Drive module
+   *
+   * @param frontLeft Swerve Module
+   * @param frontRight Swerve Module
+   * @param rearLeft Swerve Module
+   * @param rearRight Swerve Module
+   * @param kinematics Swerve drive kinematics
+   * @param gyro used for odometry and field centric driving
+   * @param maxSpeed of the wheels used to normalize wheel speeds
+   */
+  public MAXSwerve(
+      MAXModule frontLeft,
+      MAXModule frontRight,
+      MAXModule backLeft,
+      MAXModule backRight,
+      SwerveDriveKinematics kinematics,
+      SwerveModulePosition[] modulePositions,
+      AHRS gyro,
+      double maxSpeed) {
+    m_frontLeft = frontLeft;
+    m_frontRight = frontRight;
+    m_backLeft = backLeft;
+    m_backRight = backRight;
+    m_gyro = gyro;
+    m_kinematics = kinematics;
+    m_ModulePositions = modulePositions;
+    m_maxSpeed = maxSpeed;
+    m_odometry = new SwerveDriveOdometry(kinematics, m_gyro.getRotation2d(), modulePositions);
+  }
+
+  private ChassisSpeeds m_chassisSpeed = new ChassisSpeeds();
+
+  @Override
+  public void initSendable(SendableBuilder builder) {
+    super.initSendable(builder);
+    // builder.addDoubleProperty("Velocity vx", () -> m_chassisSpeed.vxMetersPerSecond, null);
```

Comment:
Please remove commented out code, or add a TODO indicating what you want to do with it.